crates/lib/src/qpu/runner.rs

```diff
@@ -9,132 +6,128 @@ use num::complex::Complex32;
 use serde::{Deserialize, Serialize};
 use serde_bytes::ByteBuf;
 
+use qcs_api_client_grpc::{
+    models::controller::{
+        data_value::Value, ControllerJobExecutionResult, DataValue, EncryptedControllerJob,
+        JobExecutionConfiguration, RealDataValue,
+    },
+    services::controller::{
+        execute_controller_job_request, get_controller_job_results_request,
+        ExecuteControllerJobRequest, GetControllerJobResultsRequest,
+    },
+};
+
 use crate::executable::Parameters;
 
-use super::engagement;
-use super::rpcq::{Client, Error as RPCQError, RPCRequest};
+use super::client::{ClientGrpcError, QcsClient};
+
+pub(crate) fn params_into_job_execution_configuration(
+    params: &Parameters,
+) -> JobExecutionConfiguration {
+    let memory_values = params
+        .iter()
+        .map(|(str, value)| {
+            (
+                str.as_ref().into(),
+                DataValue {
+                    value: Some(Value::Real(RealDataValue {
+                        data: value.to_vec(),
```

Comment:
Not sure if `value: &Vec<_>`. If so, there's a chance `value.clone()` would be more performant here.

Reply:
Can you point me to something to read on why cloning is more performant?